### DISCUSSION TITLE
Device logo selection with helper integrations

### DISCUSSION THREAD

**Discussion by dougiteixeira** [home-assistant/frontend#17121] [Configuration] [answered]
With the inclusion of helper entities in the source entity device, some devices have used the helper integration logo instead of the original integration (https://github.com/home-assistant/core/issues/95601).

Thinking about solving this, at least for the helper integrations, I think **the device logo selection could ignore helper integrations**, that way it would avoid this confusion.

This is already a known and unsolved problem for the frontend, it occurs with other integrations as well. Here is the issue history:
* https://github.com/home-assistant/frontend/issues/12123
* https://github.com/home-assistant/frontend/issues/14620
* https://github.com/home-assistant/frontend/issues/14163
* https://github.com/home-assistant/core/issues/95601

There was an attempt to improve this in PR https://github.com/home-assistant/frontend/pull/14762, but there was no consensus for approval.

I believe that this is not the definitive solution for all devices with multiple integrations, but it would solve it for helpers.

**Comment by Devil-of-Chaos** [comment]
Yes I want to set this on my own. I have many Shelly devices and want them to shown as Shelly devices because they are. But I have FritzBox too and there I only have Internet Access to change but all the shellys are changing between this icons differently!

**Comment by dougiteixeira** [comment]
Related Architectural Discussion and PR:
* https://github.com/home-assistant/architecture/discussions/936
* https://github.com/home-assistant/core/pull/95641

**Comment by dougiteixeira** [comment, marked as answer]
I believe https://github.com/home-assistant/frontend/pull/17105 solves this.

**Comment by Devil-of-Chaos** [comment]
No this is only for Helpers I mean this, they are all Shelly devices so I want to set Shelly as Main Integration.
And if I restart HA they are mixed again.
<img width="1661" alt="Bildschirmfoto 2023-07-07 um 16 20 19" src="https://github.com/home-assistant/frontend/assets/1410820/e70215b4-c20b-41f7-b4f1-02f07389b469">

**Reply by dougiteixeira** [reply]
This feature request was just the mess the helpers were making, this has been resolved.

Solution for all is being prepared at https://github.com/home-assistant/core/pull/95641.
